src/debugger/direct/directDebugAdapter.ts

```diff
@@ -0,0 +1,198 @@
+                        .then((packagerPort: number) => {
+                            this.outputLogger(`Connecting to ${packagerPort} packager port`);
+                            const attachArguments = Object.assign({}, attachArgs, {
+                                address: "localhost",
+                                port: packagerPort,
+                                restart: true,
+                                request: "attach",
+                                remoteRoot: undefined,
+                                localRoot: undefined,
+                            });
+                            super.attach(attachArguments).then(() => {
+                                this.outputLogger("The debugger attached successfully");
+                                resolve();
+                            }).catch((e) => reject(e));
+                        }).catch((e) => reject(e));
+            })
+            .catch((err) => {
+                this.outputLogger("An error occurred while attaching to the debugger. " + err.message || err, true);
+                this.cleanUp();
+            });
+        }));
+    }
+
+    public disconnect(args: DebugProtocol.DisconnectArguments): void {
+        this.cleanUp();
+        super.disconnect(args);
+    }
+
+    private initializeSettings(args: any): Q.Promise<any> {
+        if (!this.isSettingsInitialized) {
+            let chromeDebugCoreLogs = getLoggingDirectory();
+            if (chromeDebugCoreLogs) {
+                chromeDebugCoreLogs = path.join(chromeDebugCoreLogs, "ChromeDebugCoreLogs.txt");
+            }
+            let logLevel: string = args.trace;
+            if (logLevel) {
+                logLevel = logLevel.replace(logLevel[0], logLevel[0].toUpperCase());
+                logger.setup(Logger.LogLevel[logLevel], chromeDebugCoreLogs || false);
+            } else {
+                logger.setup(Logger.LogLevel.Log, chromeDebugCoreLogs || false);
+            }
+
+            if (!args.sourceMaps) {
+                args.sourceMaps = true;
+            }
+
+            const projectRootPath = getProjectRoot(args);
+            return ReactNativeProjectHelper.isReactNativeProject(projectRootPath)
+            .then((result) => {
+                if (!result) {
+                    throw ErrorHelper.getInternalError(InternalErrorCode.NotInReactNativeFolderError);
+                }
+                this.projectRootPath = projectRootPath;
+                this.remoteExtension = RemoteExtension.atProjectRootPath(this.projectRootPath);
+                const version = getExtensionVersion();
+
+                // Start to send telemetry
+                (this._session as any).getTelemetryReporter().reassignTo(new RemoteTelemetryReporter(
+                    "react-native-tools", version, Telemetry.APPINSIGHTS_INSTRUMENTATIONKEY, this.projectRootPath));
+
+                this.isSettingsInitialized = true;
+
+                return void 0;
```

Comment:
Should we resolve this promise instead of returning `void 0`?

Reply:
I can't use there a `resolve` method, because the type of promise is Q.Promise, not raw js Promise. According this https://github.com/kriskowal/q#the-end I should return a value.